src/core_ocean/shared/mpas_ocn_equation_of_state_wright.F

```diff
@@ -0,0 +1,717 @@
+         dp0dS,            &! d(p0)/d(salinity)
+         drhodT,           &! derivative of density with respect to temperature
+         drhodS             ! derivative of density with respect to salinity
+
+      !-----------------------------------------------------------------
+
+      !*** initialize error flag
+
+      err = 0
+
+
+      allocate(tracerTemp(nVertLevels, nCells))
+      allocate(tracerSalt(nVertLevels, nCells))
+      allocate(boussinesqPres(nVertLevels, nCells))
+
+      if (displacementType == 'surfaceDisplaced') then
+         call compute_surface_displaced_T_S(nVertLevels, nCells, indexT,      &
+                                            indexS, tracersSurfaceLayerValue, &
+                                            tracerTemp, tracerSalt)
+      else
+         call compute_bounded_T_S(nVertLevels, nCells, indexT, indexS, tracers, &
+                                  tracerTemp, tracerSalt)
+      end if
+      call compute_boussinesq_p(nVertLevels, nCells, kDisplaced,      &
+                                displacementType, zMid, maxLevelCell, &
+                                boussinesqPres)
+
+
+#ifdef MPAS_OPENACC
+      !$acc enter data copyin(density, boussinesqPres, tracerTemp, tracerSalt, &
+      !$acc&                  thermalExpansionCoeff,  &
+      !$acc&                  salineContractionCoeff)
+      !$acc parallel loop gang vector collapse(2)     &
+      !$acc&   present(density, boussinesqPres, tracerTemp, tracerSalt, &
+      !$acc&           thermalExpansionCoeff,         &
+      !$acc&           salineContractionCoeff)
+#else
+      !$omp parallel
+      !$omp do schedule(runtime) &
+      !$omp private(k, S, T, p, T2, T3, p0, dp0dT, dp0dS, lamda0, dlambda0dT, dlambda0dS, &
+      !$omp         alpha0, dalpha0dT, dalpha0dS, denom, denom2, drhodT, drhodS)
+#endif
+
+
+      do iCell=1,nCells
+         do k=1,nVertLevels
+
+            S = tracerSalt(k, iCell)
+            T = tracerTemp(k, iCell)
+            p = boussinesqPres(k, iCell)
+
+            T2 = T*T
+            T3 = T*T2
+
+            p0 = b0 + b1*T + b2*T2 + b3*T3 + b4*S + b5*T*S
+            dp0dT = b1 + 2.0_RKIND*b2*T + 3.0_RKIND*b3*T2 + b5*S
+            dp0dS = b4 + b5*T
+
+            lambda0 = c0 + c1*T + c2*T2 + c3*T3 + c4*S + c5*T*S
+            dlambda0dT = c1 + 2.0_RKIND*c2*T + 3.0_RKIND*c3*T2 + c5*S
+            dlambda0dS = c4 + c5*T
+
+            alpha0 = a0 + a1*T + a2*S
+            dalpha0dT = a1
+            dalpha0dS = a2
+
+            denom = 1.0_RKIND/(lambda0 + alpha0*(p + p0))
+            denom2 = denom*denom
+
+            drhodT = (lambda0*dp0dT - (p + p0)*(dlambda0dT + (p + p0)*dalpha0dT))*denom2
+            drhodS = (lambda0*dp0dS - (p + p0)*(dlambda0dS + (p + p0)*dalpha0dS))*denom2
+
+            density(k, iCell) = (p + p0)*denom
+            thermalExpansionCoeff(k, iCell) = -drhodT/density(k, iCell)
+            salineContractionCoeff(k, iCell) = drhodS/density(k, iCell)
```

Comment:
@cbegeman and @dengwirda, finally, here's the EOS plus expansion/contraction coefficients.